### PR TITLE
Animal cruelty nerf - Microwaves no longer instantly gib living creatures + Microwaves now heat in real-time

### PR DESCRIPTION
**Pull request by deathride58**
## About the PR
TL;DR: Hammy and other holdable critters can now be practically revived if microwaved, but still die as a result.

This PR is straight-forward, and does exactly what it says in the title. This also tweaks the level of heat that microwaves apply to objects within, as microwaves previously didn't affect object temperature much at all. The side effect of this is that this means microwaves are now a practical way to reheat Hammy if they're cold, just like the hamster reanimation story!

## Why / Balance
This primarily addresses a pretty grating issue: Players stuffing Hammy in the microwave over the tiniest of perceived slights. This is excruciatingly awful for players who wanna play Hammy, as being microwaved meant being gibbed, which is, effectively, an *instant* round removal. Needless to say, this is incredibly toxic.

So this PR makes a slight rework to how microwaves work, and how mobs react to being microwaved. More specifically, mobs no longer have an explicit reaction to being microwaved, and microwaves now heat their contents in real-time. This PR also adds a 100x multiplier to the heat that microwaves apply to objects (Chemical contents are unaffected), as before, they surprisingly didn't affect object temperature much at all, due to the way thermodynamic heat resistance is modeled.

This ultimately means that microwave cooking time is a factor in microwaving mobs; Rats take just a few seconds to kill in the microwave, whilst Hammy takes about 30 seconds to fall into crit from being microwaved! The amount of time a mob takes to properly microwave depends on a combination of their specific heat and physical mass.

Hammy in particular benefits the most from this change. It's a *lot* easier to revive a corpse than it is a pile of gibs, which means it's now very much practical to revive Hammy (whether that's through admin tools or various ingame methods) after a trip in the microwave.

## Technical details
This PR removes `OnBeingMicrowaved` from the server's `BodySystem.cs`, tweaks the amount of heat applied to objects within microwaves, and makes microwaves do `AddTemperature()` every update. Straight-forward and self-explanatory.

While we're at it, we also moved the heat multiplier to a vareditable variable within `MicrowaveComponent`. Yknow, just in case any badmin wishes to casually create a thermonuclear microwave.

## Media

https://github.com/space-wizards/space-station-14/assets/6356337/964adc10-043c-46e0-814d-9c5b110a5145

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes
N/A

**Changelog**

:cl: Bhijn and Myr
- tweak: Mobs are no longer instantly gibbed when microwaved.
- tweak: Microwaves now heat their contents in real-time.
- tweak: Microwaves are now capable of properly heating their contents, such that microwaveable mobs will still reliably die as expected when microwaved. (They just won't be gibbed.)

### CODE REVIEW

**Comment by LankLTE** [comment]
Honestly this kind of feels like a change just towards killing hamlet which could / should probably be fixed some other way (ie letting them exit the microwave). Probably just being reluctant to change it but microwaves gibbing animals is kind of funny / iconic and I also think it's a shame to see the bloody microwave sprite go.

**Comment by deathride58** [comment]
Something we thought of but ultimately decided against is adding a critical temperature point, at which point the mob within the microwave gibs. However, would cause the problematic behavior of people gibbing Hammy over the tiniest of perceived slights to still persist, albeit in a more prolonged fashion that, frankly, would feel even worse than what the current microwave behavior encourages.
Something that *might* work is adding that behavior but with a tag that can be used to blacklist specific mobs from microwave gibbing, but in a systems-driven game like this, it often ends up feeling incredibly obtrusive for specific mobs/objects to be outright exempt from hyperspecific mechanics.

**Comment by lzk228** [comment]
HamletBurger real

**Comment by deltanedas** [comment]
> 100x multiplier to the heat that microwaves apply to objects,

wont this make microwaves stupidly op for chem

**Comment by deathride58** [comment]
Nope, chem's completely unaffected by this; microwaves affect TemperatureComponents and SolutionContainerManagerComponents separately, and the former simply does not affect the latter.

**Comment by ps3moira** [comment]
It's pretty funny watching Hamlet explode in a microwave instantly, maybe make a fix where they can exit like LankLTE said.

**Comment by metalgearsloth** [comment]
Long-term would be better to eventually gib but I think this is fine.
